atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/AbstractDbWriteTable.java

```diff
@@ -135,4 +138,39 @@ public void delete(List<Entry<Cell, Long>> entries) {
                 + "  AND m.ts = ?",
                 args);
     }
+
+    @Override
+    public void delete(RangeRequest range) {
+        String prefixedTableName = prefixedTableNames.get(tableRef);
+        StringBuilder query = new StringBuilder();
+        query.append(" /* DELETE_RANGE (").append(prefixedTableName).append(") */ ");
+        query.append(" DELETE FROM ").append(prefixedTableName).append(" m ");
+
+        // add where clauses to the query
+        byte[] start = range.getStartInclusive();
+        byte[] end = range.getEndExclusive();
+        Collection<byte[]> cols = range.getColumnNames();
+        List<Object> args = Lists.newArrayListWithCapacity(2 + cols.size());
+        List<String> whereClauses = Lists.newArrayListWithCapacity(3);
+        if (start.length > 0) {
+            whereClauses.add(range.isReverse() ? "m.row_name <= ?" : "m.row_name >= ?");
```

Comment:
I don't believe all code paths here are covered by the new tests, need to try it out with range reversed.